src/firefly/java/edu/caltech/ipac/firefly/server/catquery/BaseGator.java

```diff
@@ -72,11 +72,15 @@ public DataGroup fetchDataGroup(TableServerRequest req) throws DataAccessExcepti
         }
     }
 
-    protected File loadDataFile(TableServerRequest request) throws DataAccessException {
-        CatalogRequest req = QueryUtil.assureType(CatalogRequest.class, request);
+    protected File loadDataFile(TableServerRequest request) throws DataAccessException, IOException {
+        CatalogRequest req = getCatalogRequest(request);
         return searchGator(req);
     }
 
+    protected CatalogRequest getCatalogRequest(TableServerRequest request) throws DataAccessException, IOException {
```

Comment:
Looks good.  But, does IOException need to be thrown?  Same for `loadDataFile`.

Reply:
I tried removing it but editor shows me error at [this line](https://github.com/IPAC-SW/irsa-ife/blob/1fee3eb523c74424977a616cce9ec06801f374af/src/irsa/java/edu/caltech/ipac/irsa/sha/persistence/IrsEnhancedQuery.java#L94) and [this line](https://github.com/IPAC-SW/irsa-ife/blob/1fee3eb523c74424977a616cce9ec06801f374af/src/irsa/java/edu/caltech/ipac/irsa/sha/persistence/IrsEnhancedQuery.java#L82)

@loitly Should I remove these catch blocks in the methods that use getCatalogRequest?

Reply:
`UnsupportedEncodingException` is not thrown anymore.  You've remove the encoding part.  You can remove this `catch` block.

`inspectRequest` is not used by anyone.  I don't see any reason for it being there.  It can be removed as well.

Reply:
Ok thanks - I'll remove them.